app/controllers/reviews_controller.rb

```diff
@@ -1,2 +1,41 @@
 class ReviewsController < ApplicationController
+  before_action :find_chalet, only: [:new, :create]
+  before_action :find_user, only: [:new, :create]
+
+  def new
+    @review = Review.new
+  end
+
+  def create
+    @review = Review.new(review_params)
+    @review.chalet = @chalet
+    @review.user = @user
+    if @review.save
+      redirect_to chalet_path(@chalet)
+    else
+      render :new
```

Comment:
The **new** view does not exist in the ReviewsController, as we are only displaying the reviews within the chalets#show view, so I think the way to handle this is _render "chalets/show"_

Might need to recode this when we get to testing